rust/tw_keypair/src/ed25519/modifications/cardano/extended_keypair.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright © 2017-2023 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+use crate::ed25519::modifications::cardano::{
+    extended_private::ExtendedPrivateKey, extended_public::ExtendedPublicKey,
+};
+use crate::ed25519::{signature::Signature, Hasher512};
+use crate::traits::{KeyPairTrait, SigningKeyTrait, VerifyingKeyTrait};
+use crate::Error;
+use tw_encoding::hex;
+use zeroize::Zeroizing;
+
+/// Represents an `ed25519` extended key pair that is used in Cardano blockchain.
+pub struct ExtendedKeyPair<H: Hasher512> {
+    private: ExtendedPrivateKey<H>,
+    public: ExtendedPublicKey<H>,
+}
+
+impl<H: Hasher512> KeyPairTrait for ExtendedKeyPair<H> {
+    type Private = ExtendedPrivateKey<H>;
+    type Public = ExtendedPublicKey<H>;
+
+    fn public(&self) -> &Self::Public {
+        &self.public
+    }
+
+    fn private(&self) -> &Self::Private {
+        &self.private
+    }
+}
+
+impl<H: Hasher512> SigningKeyTrait for ExtendedKeyPair<H> {
+    type SigningMessage = Vec<u8>;
+    type Signature = Signature;
+
+    fn sign(&self, message: Self::SigningMessage) -> Result<Self::Signature, Error> {
```

Comment:
Let's also have alias in general for such things: `MyConcreteResult = Result<Self::Signature, Error>`

Reply:
Done